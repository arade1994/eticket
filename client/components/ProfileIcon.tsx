interface Props {
  firstName: string;
  lastName: string;
}

const ProfileIcon = ({ firstName, lastName }: Props) => {
  return (
    <div
      style={{
        border: "1px solid white",
        borderRadius: "50%",
        width: 40,
        height: 40,
        display: "flex",
        background: "black",
        color: "lightsteelblue",
      }}
    >
      <p style={{ margin: "auto" }}>{`${firstName[0]}${lastName[0]}`}</p>
    </div>
  );
};

export default ProfileIcon;
