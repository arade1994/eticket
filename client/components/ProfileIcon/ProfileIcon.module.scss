.profileIcon {
  width: 40px;
  height: 40px;
  display: flex;
  background: black;
  color: lightsteelblue;
  border: 1px solid white;
  border-radius: 50%;
}
