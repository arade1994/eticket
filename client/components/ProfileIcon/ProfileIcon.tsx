import classes from "./ProfileIcon.module.scss";

interface Props {
  firstName: string;
  lastName: string;
}

const ProfileIcon = ({ firstName, lastName }: Props) => {
  return (
    <div className={classes.profileIcon}>
      <p style={{ margin: "auto" }}>{`${firstName[0]}${lastName[0]}`}</p>
    </div>
  );
};

export default ProfileIcon;
