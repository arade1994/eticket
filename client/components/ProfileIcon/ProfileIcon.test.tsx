import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";

import ProfileIcon from "./ProfileIcon";

describe("<ProfileIcon />", () => {
  test("it should render profile icon component as expected", () => {
    const { baseElement } = render(
      <ProfileIcon firstName="Mick" lastName="Jones" />
    );

    expect(baseElement).toMatchSnapshot();

    const initials = screen.getByText("MJ");
    expect(initials).toBeDefined();
  });
});
