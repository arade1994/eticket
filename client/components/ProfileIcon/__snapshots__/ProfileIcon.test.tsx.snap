// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProfileIcon /> > it should render profile icon component as expected 1`] = `
<body>
  <div>
    <div
      class="_profileIcon_a7b94c"
    >
      <p
        style="margin: auto;"
      >
        MJ
      </p>
    </div>
  </div>
</body>
`;
